utils/node_local_test.py

```diff
@@ -1072,10 +1079,21 @@ def run_daos_client_cmd_pil4dfs(self, cmd, check=True):
         cmd_env['DAOS_AGENT_DRPC_DIR'] = self.conf.agent_dir
         cmd_env['D_IL_REPORT'] = '1'
         cmd_env['LD_PRELOAD'] = join(self.conf['PREFIX'], 'lib64', 'libpil4dfs.so')
+        if container is not None:
+            # Create a temporary directory for the mount point, this will be removed as it goes out
+            # scope so keep as a local for the rest of the function.
+            # pylint: disable-next=consider-using-with
+            tmp_dir = tempfile.TemporaryDirectory(prefix='pil4dfs_mount')
+            cwd = tmp_dir.name
```

Comment:
You might consider `tempfile.mkdtemp`, which I think pylint doesn't complain about
https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp
Because:
> The user of [mkdtemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp) is responsible for deleting the temporary directory and its contents when done with it.

Though I haven't verified

Reply:
Well, we do want the directory removed.  The only reason we need to suppress this is because doing it properly would require re-indenting the rest of the function and more testing, something that would make sense when the code was written but for now it's easier to move forward with the suppression.